phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixEmbeddedDriver.java

```diff
@@ -517,20 +570,46 @@ public ConnectionInfo(String zookeeperQuorum, Integer port, String rootNode) {
          * @param other The instance to copy
          */
         public ConnectionInfo(ConnectionInfo other) {
-            this(other.zookeeperQuorum, other.port, other.rootNode, other.principal, other.keytab);
+            this(other.quorum, other.port, other.rootNode, other.principal, other.keytab);
         }
 
         public ReadOnlyProps asProps() {
             Map<String, String> connectionProps = Maps.newHashMapWithExpectedSize(3);
-            if (getZookeeperQuorum() != null) {
-                connectionProps.put(QueryServices.ZOOKEEPER_QUORUM_ATTRIB, getZookeeperQuorum());
-            }
-            if (getPort() != null) {
-                connectionProps.put(QueryServices.ZOOKEEPER_PORT_ATTRIB, getPort().toString());
+            if (this.isHRPCBootstrap() && getQuorum() != null) {
+                final String[] masters = getQuorum().split(",");
```

Comment:
While the enum suggestion is cleaner, I think pushing this "bootstrapping" into separate classes for each implementation is a pretty standard object-oriented pattern to follow. (e.g. make an interface for PhoenixRuntimeBootstrap with two implementations: one for ZK and one for MasterRegistry).

PhoenixEmbeddedDriver.java is a big mess already and I think pushing even more logic into it will make it worse.

Reply:
+1.